common/flatpak-run.c

```diff
@@ -381,7 +383,9 @@ flatpak_run_add_environment_args (FlatpakBwrap    *bwrap,
                               "--dev", "/dev",
                               NULL);
 
-      if (context->devices & FLATPAK_CONTEXT_DEVICE_USB)
+      if (context->devices & FLATPAK_CONTEXT_DEVICE_USB &&
+	  !(g_hash_table_size (context->enumerable_usb_devices) != 0 &&
+	    flatpak_has_usb_portal ()))
```

Comment:
This kind of magic is a bad idea. I get the idea but let's not create even more ad-hoc fallback mechanisms.

Reply:
the bad idea is making things more complicated than it should. If the USB portal is request then any USB static permissions should be removed. And these permissions exist for two case: app doesn't support portal, or portal isn't available. In both case it is opt-in.

`--nousb=all` should be able to disable requesting the portal (it's not implemented as it's not that easy).